routers/api/v1/repo/issue_pin.go

```diff
@@ -45,6 +45,8 @@ func PinIssue(ctx *context.APIContext) {
 	if err != nil {
 		if issues_model.IsErrIssueNotExist(err) {
 			ctx.NotFound()
+		} else if issues_model.IsErrIssueMaxPinReached(err) {
+			ctx.Error(http.StatusBadGateway, "MaxPinReached", err)
```

Comment:
StatusBadGateway is for reverse-proxy only, Gitea never uses this status code.

Application related error code could be either 40x or 500 (InternalServerError)

Here 40x seems better.

Reply:
Indeed, a generic 400 (Bad Request) seems suitable.